backend/src/main/java/eventlistener/service/notificaionuser/NotificationUserService.java

```diff
@@ -38,19 +42,30 @@ public NotificationUser addUser(NotificationUserDTO userDTO) {
         if(eventService.eventsExist(userDTO.getListenEvents())){
             NotificationUser newUser = notificationUserMapper.mapDTO(userDTO);
             NotificationUser userWithID = notificationUserRepo.save(newUser);
-            eventService.addUserToEvents(userDTO.getListenEvents(), userWithID);
+            eventService.addUserToEvents(userDTO.getListenEvents(), userWithID.getId());
```

Comment:
```suggestion
            eventService.addUserToEvents(user);
```
and then let the eventService extract the data it needs from the user